src/storage/src/keyspace.rs

```diff
@@ -111,7 +143,18 @@ impl<S: StateStore> Keyspace<S> {
     /// Gets from the keyspace with the `prefixed_key` of given key.
     /// The returned value is based on a snapshot corresponding to the given `epoch`
     pub async fn get(&self, key: impl AsRef<[u8]>, epoch: u64) -> StorageResult<Option<Bytes>> {
-        self.store.get(&self.prefixed_key(key), epoch).await
+        self.store.get(&self.prefixed_key(key), epoch, None).await
+    }
+
+    pub async fn get_with_vnode(
+        &self,
+        key: impl AsRef<[u8]>,
+        epoch: u64,
+        vnode: VirtualNode,
+    ) -> StorageResult<Option<Bytes>> {
+        self.store
+            .get(&self.prefixed_key(key), epoch, Some(vnode))
+            .await
```

Comment:
This function uses the given `vnode` instead of `self.vnode`. What scenario should it be used?

Reply:
It is used when an executor does point-get with vnode.

Reply:
I think point get can already be optimized by bloom filter (only 0.01 false positive currently). Maybe we don't need vnode for it. But it would also be okay to use vnode to do some sanity check -- e.g. executors should not point get keys out of its vnode range.